pkgs/development/compilers/mlkit/default.nix

```diff
@@ -0,0 +1,29 @@
+{ stdenv, fetchFromGitHub, autoreconfHook, mlton }:
+
+stdenv.mkDerivation rec {
+  pname = "mlkit";
+  version = "4.5.0";
+
+  src = fetchFromGitHub {
+    owner = "melsman";
+    repo = "mlkit";
+    rev = "v${version}";
+    sha256 = "0fc0y40qphn02857fv2dvhwzzsvgixzchx9i6i0x80xfv7z68fbh";
+  };
+
+  nativeBuildInputs = [ autoreconfHook mlton ];
+
+  enableParallelBuilding = true;
+
+  buildPhase = ''
+    make mlkit
+    make mlkit_libs
+  '';
```

Comment:
Replacing the buildPhase with makeFlags unfortunately doesn't work, likely because the Makefile has imperfect dependencies. make mlkit must finish completely before running make mlkib_libs.

Reply:
That produces build errors, likely for the same reason `makeFlags` doesn't work (bad dependencies in the upstream Makefile).

Reply:
Works fine here when I disable parallel builds. The makefile probably doesn't support proper parallel builds. The reason it works with the `buildPhase` as you had it is that it does not pass the `-j` flag to make and thus does not do a parallel build.

Reply:
Indeed.  I have made the suggested change.